autoscheduler/scraper/models/grades.py

```diff
@@ -0,0 +1,75 @@
+""" Heavily based off of Good Bull Schedules """
+
+from typing import Dict
+from django.db import models
+
+class GradeManager(models.Manager):
+    """ Connects to the Grades models so we can call
+        Grades.object.instructor_performance
+    """
+
+    def instructor_performance(
+            self, dept: str, course_num: str, instructor: str
+    ) -> Dict[str, float]:
+        """ Aggregates all of the GPA's / sum of each grade type into one object
+            so we can quickly get how an instructor performed in past sections
+            for a given course
+
+            Returns a dictionary of string-integer, pairs, where str is the attribute
+            (i.e. gpa), and float is the according value (is an integer for grade counts)
+        """
+
+        return (
+            # prefetch_related retrieves all of the related objects in a single query
+            self.prefetch_related("section").filter(
+                section__subject=dept,
+                section__course_num=course_num,
+                section__instructor=instructor,
+            ).aggregate(
+                gpa=models.Avg("gpa"), # Averages all of the GPA's together
+
+                # We want a sum so we can more-easily calculate the percentage
+                # Sums the count of each grade
+                A=models.Sum("A"),
+                B=models.Sum("B"),
+                C=models.Sum("C"),
+                D=models.Sum("D"),
+                F=models.Sum("F"),
+                I=models.Sum("I"),
+                S=models.Sum("S"),
+                U=models.Sum("U"),
+                Q=models.Sum("Q"),
+                X=models.Sum("X"),

```

Comment:
Where are we going to calculate percentages? I assume in the api route (we could cache it after calculating once if we do so), but just making sure we're on the same page

Reply:
Honestly I was thinking that the frontend would deal with the percentages and we just give them the sums, but we can calculate percentages if you think that's better. My thought process was that giving them the sums would provide the frontend with a bit more flexibility with what they want to do with it.

Reply:
True, calculating it client-side will be faster for us and provide more flexibility with how they display grade distributions (we could show total students, number of students who got each letter grade, etc), so that seems like the way to go.